File: uva11799_Horror-Dash/uva11799.py
# python 3.5.2
# one linear scan to find the max value

strTestCase = input()
iTestCase = int(strTestCase)
lStudentSpeed = []

for i in range(1,iTestCase+1):
	iMaxSpeed = 0;
	strStudentSpeed = input()
	lStudentSpeed = strStudentSpeed.split(" ")
	for speed in lStudentSpeed:
		if (iMaxSpeed < int(speed)):
			iMaxSpeed = int(speed)
	print ("Case %d: %d"%(i,iMaxSpeed))

File: uva11799_Horror-Dash/uva11799.cpp
// One linear scan to find the max value
#include <stdio.h>

int main()
{
	int testCase;
	int studentSpeed;
	int numberOfStudents;
	int Case = 0;
	
	scanf("%d",&testCase);
	while (testCase)
	{
		Case = Case + 1;
		int maxSpeed = 0;
		
		scanf("%d",&numberOfStudents);
		for (int i = 0; i < numberOfStudents; i++)
		{
			scanf("%d",&studentSpeed);
			if (studentSpeed > maxSpeed)
				maxSpeed = studentSpeed;
		}
		printf("Case %d: %d\n",Case,maxSpeed);
		testCase--;
	}
	return 0;
}
